events/views.py

```diff
@@ -45,3 +58,175 @@ def update_event(request, pk):
+            if form.cleaned_data.get('date_time_end'):
+                form_data['date_time_end'] = form.cleaned_data.get('date_time_end').strftime("%Y-%m-%dT%H:%M")
+            if form.cleaned_data.get('date_time_start'):
+                form_data['date_time_start'] = form.cleaned_data.get('date_time_start').strftime("%Y-%m-%dT%H:%M")
+            result.append(form_data)
+        return result
+
+    @staticmethod
+    def get_formset_participant_date(request):
+        result = []
+        keys = list(filter(lambda x: "participants-" in x, request.POST.keys()))[4:-1]
+        for index, key in enumerate(keys):
+            form_data = {}
+            form_data['id'] = index
+            form_data['email'] = request.POST[key]
+            result.append(form_data)
+        return result
+
+    def saving_all_optional_meeting_dates(self, event_creator, event_instance, optional_meetings_formset):
+        _ = list(map(lambda form: self.adding_event_creator(form, event_creator), optional_meetings_formset))
+
+        # add the event time to the optional meeting dates
+        OptionalMeetingDates(
+            event_creator_id=event_creator,
+            date_time_start=event_instance.date_time_start,
+            date_time_end=event_instance.date_time_end).save()
+
+    def check_optional_meeting_dates_formset(self, request, event_instance, event_creator, optional_meetings_formset):
+        if optional_meetings_formset.is_valid():
+            is_meeting_formset_invalid = False
+            for form in optional_meetings_formset:
+                is_meeting_formset_invalid = self.check_dates_constraint(form, event_instance, request)
+                if is_meeting_formset_invalid:
+                    break
+            if is_meeting_formset_invalid:
+                event_instance.delete()
+                return False
+            else:
+                self.saving_all_optional_meeting_dates(event_creator, event_instance, optional_meetings_formset)
+        else:
+            event_instance.delete()
+            return False
+        return True
+
+    @staticmethod
+    def check_participant_formset(request, event_instance, meeting_participants_formset):
+        if meeting_participants_formset.is_valid():
+            for form in meeting_participants_formset:
+                if form.is_valid():
+                    try:
+                        participant_email = form.cleaned_data.get('participant_email')
+                        if participant_email:
+                            user_instance = User.objects.get(email=participant_email)
+                            participant = EventParticipant(
+                                event_id=event_instance, user_id=user_instance, is_creator=False
+                            )
+                            participant.save()
+                    except User.DoesNotExist:
+                        messages.warning(request, f"There is not user with the email: {participant_email}")
```

Comment:
please change message to "There is no* user with the email: XX"

Reply:
Done, thanks 😄